src/v2/view-builder/views/webauthn/ChallengeWebauthnAuthenticatorView.js

```diff
@@ -0,0 +1,142 @@
+
+const Body = BaseForm.extend({
+
+  title () {
+    return loc('oie.verify.webauth.title', 'login');
+  },
+
+  className: 'oie-verify-webauthn',
+
+  getUISchema () {
+    const schema = [];
+    // Returning custom array so no input fields are displayed for webauthn
+    if (webauthn.isNewApiAvailable()) {
+      const retryButton = createButton({
+        className: 'retry-webauthn button-primary default-custom-button',
+        title: loc('retry', 'login'),
+        click: () => {
+          this.getCredentialsAndSave();
+        }
+      });
+
+      schema.push({
+        View: ChallengeWebauthnInfoView,
+      }, {
+        View: retryButton,
+      });
+    }
+    return schema;
+  },
+
+  remove () {
+    BaseForm.prototype.remove.apply(this, arguments);
+    if (this.webauthnAbortController) {
+      this.webauthnAbortController.abort();
+      this.webauthnAbortController = null;
+    }
+  },
+
+  noButtonBar: true,
+
+  modelEvents: {
+    'error': '_stopVerification'
+  },
+
+  getCredentialsAndSave () {
+    this.clearErrors();
+    this._startVerification();
+    this.webauthnAbortController = new AbortController();
+    const authenticatorData = this.options.appState.get('currentAuthenticatorEnrollment');
+    const allowCredentials = [];
+    const authenticatorEnrollments = this.options.appState.get('authenticatorEnrollments').value || [];
+    authenticatorEnrollments.forEach((enrollement) => {
+      if (enrollement.type === 'security_key') {
+        allowCredentials.push({
+          type: 'public-key',
+          id: CryptoUtil.strToBin(enrollement.credentialId),
+        });
+      }
+    });
+    const options = {
+      allowCredentials,
+      userVerification: authenticatorData.contextualData.challengeData.userVerification,
+      challenge: CryptoUtil.strToBin(authenticatorData.contextualData.challengeData.challenge),
+    };
+    navigator.credentials.get({
+      publicKey: options,
+      signal: this.webauthnAbortController.signal
+    }).then((assertion) => {
+      this.model.set({
+        credentials : {
+          clientData: CryptoUtil.binToStr(assertion.response.clientDataJSON),
+          authenticatorData: CryptoUtil.binToStr(assertion.response.authenticatorData),
+          signatureData: CryptoUtil.binToStr(assertion.response.signature),
+        }
+      });
+      this.saveForm(this.model);
+    }, (error) => {
+      // Do not display if it is abort error triggered by code when switching.
+      // this.webauthnAbortController would be null if abort was triggered by code.
+      if (this.webauthnAbortController) {
+        this.model.trigger('error', this.model, { responseJSON: { errorSummary: error.message } });
+      }
+    }).finally(() => {
+      // unset webauthnAbortController on successful authentication or error
+      this.webauthnAbortController = null;
+    });
+  },
+
+  postRender: function () {
+    _.defer(() => {
```

Comment:
@haishengwu-okta  I need to use the defer coz I can add this in the View extending BaseFactorView.

Any better idea. I was thinking of firing an event on appState/mode on post render.

Reply:
- trigger event on model would probably work.
- did you try to override `postRender` in the View and call `this.form.getCredentialsAndSave`) (similar idea from Enroll)